### PR TITLE
[Merged by Bors] - feat: port algebra.hom.group_instances

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
~~Todo: `to_additive` on the `CommGroup` instance seems to take forever and still not work.~~

### CODE REVIEW

File: Mathlib/Algebra/Hom/GroupInstances.lean [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  npow n f :=
+    { toFun := fun x => f x ^ n, map_one' := by simp, map_mul' := fun x y => by simp [mul_pow] }
+  npow_zero f := by
+    ext x
+    simp
+  npow_succ n f := by
+    ext x
+    simp [pow_succ]
+
+set_option maxHeartbeats 2000000
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Hom/GroupInstances.lean:47]: I suspect we won't want to land this

**Comment by fpvandoorn** [comment]
I'll try to improve the performance of `@[to_additive]`. I know what the bottleneck is, and it was on my list for a while, but now I'll try to do it much sooner.

**Comment by fpvandoorn** [comment]
#1060 should fix the performance issues

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**